lib/ansible/modules/cloud/amazon/lightsail.py

```diff
@@ -223,9 +264,44 @@ def create_instance(module, client, instance_name):
             resp = resp['operations'][0]
         except botocore.exceptions.ClientError as e:
             module.fail_json(msg='Unable to create instance {0}, error: {1}'.format(instance_name, e))
-        changed = True
 
-    inst = _find_instance_info(client, instance_name)
+        inst = _find_instance_info(client, instance_name)
+
+        # Wait for instance to become running
+        if wait:
+            while (wait_max > time.time()) and (inst is not None and inst['state']['name'] != "running"):
```

Comment:
Should inst ever be None without being in the process of being created at this point? Wouldn't it be worth waiting for that too? If so you could remove the call to _find_instance_info(client, instance_name) just above and change this conditional.